elements/pl-dataframe/pl-dataframe.py

```diff
@@ -0,0 +1,164 @@
+
+    return "Unknown"
+
+
+def get_pandas_dtype(s: pd.Series) -> str:
+    return s.dtype
+
+
+def prepare(element_html: str, data: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    pl.check_attribs(
+        element,
+        required_attribs=["params-name"],
+        optional_attribs=[
+            "show-index",
+            "show-header",
+            "show-dimensions",
+            "show-dtype",
+            "show-python",
+            "display-language",
+            "digits",
+        ],
+    )
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    element = lxml.html.fragment_fromstring(element_html)
+    # TODO should there be a different display varname?
+    varname = pl.get_string_attrib(element, "params-name")
+    show_index = pl.get_boolean_attrib(element, "show-index", SHOW_INDEX_DEFAULT)
+    show_header = pl.get_boolean_attrib(element, "show-header", SHOW_HEADER_DEFAULT)
+    show_python = pl.get_boolean_attrib(element, "show-python", SHOW_PYTHON_DEFAULT)
+    show_dimensions = pl.get_boolean_attrib(
+        element, "show-dimensions", SHOW_DIMENSIONS_DEFAULT
+    )
+    show_dtype = pl.get_boolean_attrib(element, "show-dtype", SHOW_DTYPE_DEFAULT)
+
+    display_language = pl.get_enum_attrib(
+        element, "display-language", DisplayLanguage, DISPLAY_LANGUAGE_DEFAULT
+    )
+
+    num_digits = pl.get_integer_attrib(element, "digits", NUM_DIGITS_DEFAULT)
+
+    if varname not in data["params"]:
+        raise KeyError(
+            f'Could not find {varname} in params. Please make sure to set params-name="{varname}" in the element.'
+        )
+
+    # Always assume that entry in params dict is serialized dataframe
+    frame = pl.from_json(data["params"][varname])
+
+    if not isinstance(frame, pd.DataFrame):
+        raise ValueError(f"Parameter name '{varname}' does not encode a dataframe.")
+
+    frame = cast(pd.DataFrame, frame)
+
+    frame_style = frame.style
+
+    # Format integers using commas every 3 digits
+    integer_column_names = frame.select_dtypes(include="int").columns
+    frame_style.format(subset=integer_column_names, thousands=",")
+
+    if num_digits is not None:
+        # Get headers for all floating point columns and style them to use the desired number of sig figs.
+        float_column_names = frame.select_dtypes(include="float").columns
+
+        # This format string displays the desired number of digits, as given by the instructor
+        # Switches between exponential and decimal notation as needed
+        frame_style.format(subset=float_column_names, formatter=f"{{:.{num_digits}g}}")
+
+    if show_dtype:
+        if display_language is DisplayLanguage.PYTHON:
+            get_dtype_function = get_pandas_dtype
+        elif display_language is DisplayLanguage.R:
+            get_dtype_function = convert_pandas_dtype_to_r
+        else:
+            assert_never(display_language)
+
+        # old agg function:
```

Comment:
What does this comment mean?

Reply:
@coatless ?

Reply:
This was a note that emphasized in the POC that the lambda function originally being used in the `.agg()` could be swapped with the new `convert_pandas_dtype_to_r()` function. The note can be removed.